File: src/main/java/hzt/aoc/day06/Part1CustomCustoms.java
package hzt.aoc.day06;

import hzt.aoc.day06.model.Group;
import hzt.collections.ListX;

import java.util.List;

public class Part1CustomCustoms extends Day06Challenge {

    public Part1CustomCustoms() {
        super("part 1", "For each group, count the number of questions to which anyone answered 'yes'. " +
                "What is the sum of those counts?. ");
    }

    @Override
    protected int calculateResult(List<Group> groups) {
        return ListX.of(groups).fold(0, (acc, a) -> acc += a.amountAnyoneAnsweredYes());
    }

    @Override
    protected String getMessage(String result) {
        return String.format("The sum of the counts in each group to which anyone answered 'yes' is: %s%n", result);
    }
}

File: src/main/java/hzt/aoc/day06/Part1CustomCustoms.kt
package hzt.aoc.day06

import hzt.aoc.day06.model.Group

class Part1CustomCustoms : Day06Challenge(
    "part 1", "For each group, count the number of questions to which anyone answered 'yes'. " +
            "What is the sum of those counts. "
) {
    override fun calculateResult(groups: List<Group>): Int {
        return groups.stream()
            .mapToInt(Group::amountAnyoneAnsweredYes)
            .sum()
    }

    override fun getMessage(result: String): String {
        return String.format("The sum of the counts in each group to which anyone answered 'yes' is: %s%n", result)
    }
}
